docs: update the wan mesh gateway page (#12291)
hashicorp/consul: website/content/docs/connect/gateways/mesh-gateway/wan-federation-via-mesh-gateways.mdx

}
```

> The [`start_join_wan`](/docs/agent/options#start_join_wan) or [`retry_join_wan`](/docs/agent/options#retry_join_wan) are only used for the legacy federation process. They must be omitted when federating Consul servers via gateways.

Reviewer: Why are we calling the non-mesh-gateway federation legacy federation? Isn't that the standard federation model?  Wouldn't this be more correct?

```suggestion
The [`start_join_wan`](/docs/agent/options#start_join_wan) or [`retry_join_wan`](/docs/agent/options#retry_join_wan) must be omitted when federating Consul servers via mesh gateways. 
```
